### ISSUE TITLE
Query about changes in scripts required other than creating the dataloader for one's own dataset and the label format

### ISSUE THREAD

**Issue by AntiLibrary5** [NVlabs/geomapnet#40]
Hi,
Thank you for your work. I had 2 queries:

1. From the ground truth files of **7Scenes** and the reference paper, and the code of the **RobotCar** dataset, it seems the label format required is a **Camera Pose Matrix** (for the **common.pose_utils.process_poses** method)? I wanted to confirm this since my own ground truth contains **[X Y Z W P Q R]** where X, Y, Z are the position coordinates and W, P, Q, R the quaternions for each image. 

2. Apart from creating ones own dataloader and adding it to **composite.py**, are there any other scripts that need some addition of code/changes. I feel **scripts/train.py** and **common/train.py** also need changes but are there any other ones?

Thank you.

**Comment by samarth-robo** [comment]
Hi @AntiLibrary5, the post information can be in whatever format you want, because only your dataset loader will have to read it. The important thing is to ensure that your dataset loader converts the pose to a standard 4x4 homogeneous transform matrix, take the first 3 rows and flatten it row-wise to make a 12-element vector. This is given as `poses_in` to `process_poses()`. 

I have written down some information [in `dataset_loaders/README.md`](https://github.com/NVlabs/geomapnet/tree/master/dataset_loaders#readme). Because some time has passed since I last used this code, it is difficult for me to provide more information unless you post specific error messages along with what you have tried.

**Comment by AntiLibrary5** [comment]
Hi,
Your reply in:
https://github.com/NVlabs/geomapnet/issues/19#issuecomment-854970503

makes it completely clear. Thank you.
